src/base/emu-i386/simx86/interp.c

```diff
@@ -1780,16 +1780,15 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 			if (V86MODE() && IOPL!=3 && !(TheCPU.cr[4] & CR4_VME)) {
 			    PC++; goto not_permitted;	/* GPF */
 			}
-			/* pop from stack without adjusting esp before A_SR_* */
-			Gen(O_POP1, _mode);
-			Gen(O_POP2, _mode, Ofs_EIP);
-			Gen(O_POP2, _mode|MNOREG);
-			if (REALADDR())
+			if (REALADDR()) {
+				/* pop from stack without adjusting esp before A_SR_* */
+				Gen(O_POP1, _mode);
+				Gen(O_POP2, _mode, Ofs_EIP);
```

Comment:
But JMP_INDIRECT can return
whatever in eax, so why not to
put things into eax and leave
EIP alone? Could even be slightly
faster.

Reply:
It's basically because the next pop for cs overwrites eax. I can avoid this if the POP2 sequence does not clobber eax but I figured it wasn't worth the trouble since it's ok to set eip here. But in principle it's possible